File: ska_frontend/src/gui/page_assistant/dtos/dto_fetch_chat_messages.ts
import { UtilsTypes } from "@/utils/core/utils_types";
import { DtoChatDetails } from "./dto_chat_details";
import { DtoChatPacket } from "./dto_chat_packet";

export class DtoFetchChatMessagesRequest {
    chat_id: number;

    constructor(obj: {
        chat_id: number,
    }) {
        this.chat_id = obj.chat_id;
    }
}

export class DtoFetchChatMessagesResponse {
    user_chat: DtoChatDetails;
    chat_packets: DtoChatPacket[];

    constructor(obj: {
        user_chat: DtoChatDetails,
        chat_packets: DtoChatPacket[],
    }) {
        this.user_chat = obj.user_chat;
        this.chat_packets = obj.chat_packets;
    }

    static fromUnknown(value: unknown): DtoFetchChatMessagesResponse {
        const obj = UtilsTypes.unknownToObject(value).unwrap();
        const chatPacketsUnknown = UtilsTypes.unknownToArray(obj.chat_packets).unwrap();
        return new DtoFetchChatMessagesResponse({
            user_chat: DtoChatDetails.fromUnknown(obj.user_chat),
            chat_packets: chatPacketsUnknown.map(DtoChatPacket.fromUnknown),
        })
    }
}

File: ska_backend/src/server/route_api/route_assistant/route_fetch_chat_messages/dto_fetch_chat_messages.rs
use serde::{Deserialize, Serialize};

use crate::domain::user::dto_chat_details::{DtoChatDetails, DtoChatPacket};

#[derive(Clone, Serialize, Deserialize)]
pub struct DtoFetchChatMessagesRequest {
    pub chat_id: i64,
}

#[derive(Clone, Serialize, Deserialize)]
pub struct DtoFetchChatMessagesResponse {
    pub user_chat: DtoChatDetails,
    pub chat_packets: Vec<DtoChatPacket>,
}
